velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -356,4 +356,52 @@ struct ArrayHasDuplicatesFunction {
   }
 };
 
+template <typename TExecParams, typename T>
+struct ArrayNormalizeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  FOLLY_ALWAYS_INLINE bool callNullFree(
+      out_type<velox::Array<T>>& result,
+      const null_free_arg_type<velox::Array<T>>& inputArray,
+      const null_free_arg_type<T> p) {
+    VELOX_USER_CHECK_GE(
+        p, 0, "array_normalize only supports non-negative p: {}", p);
+
+    // If the input array is empty, then the empty result should be returned.
+    if (inputArray.size() == 0) {
+      return true;
```

Comment:
Is the result array always empty ? Is there a way to clear the array elements always before returning ?

Reply:
I could not find a method to clear an array https://github.com/facebookincubator/velox/blob/main/velox/expression/ComplexWriterTypes.h#L84. The result is empty at the start of the function in the tests but I don't know if it actually is in production.

Reply:
@mbasmanova : Wanted to confirm this with you.

Reply:
> return true;

This means that the return is NULL. Is this the correct behavior?

Reply:
@laithsakka Laith, would you clarify what't the right way to return an empty array? It might be helpful to clarify this in the documentation as well.

Reply:
The array is empty initially